spec/lib/namespaces_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe 'Namespace' do
+  namespaces = %w[CandidateInterface ProviderInterface SupportInterface APIDocs RefereeInterface VendorAPI]
+
+  namespaces.each do |namespace|
+    describe namespace.constantize do
+      (namespaces - [namespace]).each do |other_namespace|
+        it "does not use code from the #{other_namespace} namespace" do
+          command = "grep -rnw app/**/#{namespace.underscore} -e '#{other_namespace}'"
+          usage_of_namespaced_code_in_other_namespace = `#{command}`
+          expect(usage_of_namespaced_code_in_other_namespace).to be_empty, usage_of_namespaced_code_in_other_namespace
```

Comment:
Repeat of `usage_of_namespaced_code_in_other_namespace`. Not sure that's doing anything?

Reply:
The second argument makes sure that the full `grep` output is displayed if there are multiple matches.

Without the second arg:

```
  1) Namespace CandidateInterface does not use code from the ProviderInterface namespace
     Failure/Error: expect(usage_of_namespaced_code_in_other_namespace).to be_empty
       expected `"app/controllers/candidate_interface/application_form_controller.rb:7:      @application_form_present..._again')\napp/controllers/candidate_interface/degrees/base_controller.rb:11:    ProviderInterface\n".empty?` to return true, got false
     # ./spec/lib/namespaces_spec.rb:12:in `block (5 levels) in <top (required)>'
``` 

With:

```
  1) Namespace CandidateInterface does not use code from the ProviderInterface namespace
     Failure/Error: expect(usage_of_namespaced_code_in_other_namespace).to be_empty, usage_of_namespaced_code_in_other_namespace
     
       app/controllers/candidate_interface/application_form_controller.rb:7:      @application_form_presenter = ProviderInterface::ApplicationFormPresenter.new(current_application)
       app/controllers/candidate_interface/application_form_controller.rb:21:      render_404 and return unless ProviderInterface::FeatureFlag.active?('apply_again')
       app/controllers/candidate_interface/degrees/base_controller.rb:11:    ProviderInterface
     # ./spec/lib/namespaces_spec.rb:12:in `block (5 levels) in <top (required)>'
```